### PR TITLE
Create traceability and frameproof files

### PR DESCRIPTION
**Pull request by raleyr**

### CODE REVIEW

File: codes/classical/q-ary_digits/frameproof.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# that entry instead of here.
+description: |
+  Frameproof codes are those that prevent a group of users from being able to collude to frame another user outside of the group of creating an unauthorized copy of data. Even further, the group of users cannot frame a separate user even if they know the unique codeword of the separate user \cite{doi:10.1007/3-540-44750-4_36}.
+  
+  Let us start with some definitions to then define what a c-frameproof code is. A set \(\Gamma = \{w^{(1)}, \dots, w^{(n)}\} \subseteq \{(0,1)\}^{l}\) is an ( \(l,n\) )-code. Each codeword \(w^{(i)}\) correlates to a user \(u_i\).
+  Let C be a group of users. A bit in position i is undetectable for the group C when the words assigned to the users in the group match at the the same position i.
+  The feasible set of the group C, denoted \(F(C;\Gamma)\) or \(F(C)\), for some user u in C contains all of the codewords that match the groups set of undetectable bits. Finally, if every subset \(S \subset \Gamma\) of size at most c satisfies \(F(S)\cap\Gamma = S\), then \(\Gamma\) is a c-frameproof code \cite{doi:10.1007/3-540-44750-4_36}.
+
```
Comment (valbert4) [inline codes/classical/q-ary_digits/frameproof.yml:37]: c, i, etc should be in \(...\)

File: codes/classical/q-ary_digits/frameproof.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+# Delete instructional comments when submitting
+
+# code id, physical, logical are all lower case
+code_id: frameproof
+physical: q-ary_digits
+logical: q-ary_digits
+
+# Apostrophes are denoted by two apostrophe characters, i.e., ''
+# Code title (SINGULAR) + first reference(s) (optional).
+name: 'Frameproof'
```
Comment (valbert4) [inline codes/classical/q-ary_digits/frameproof.yml:26]: ...code

File: codes/classical/q-ary_digits/frameproof.yml [hunk 3]
```diff
@@ -0,0 +1,120 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+# Use UTF-8 unicode encoding
```
Comment (valbert4) [inline codes/classical/q-ary_digits/frameproof.yml:6]: Delete all instructional comments

File: codes/classical/q-ary_digits/frameproof.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+realizations:
+  # List and explain the different "domains" of realizations in list items.
+  # Group closely related realizations into a single item. It's OK to have a
+  # single list item. Each item should more or less correspond to what would be
+  # said in a paragraph.
+  - 'Frameproof codes are utilized in digital fingerprinting and watermarking \cite{doi:10.1137/S0895480101384633}.'
+
+notes:
+  - 'Any c-frameproof code must have at least length c \cite{doi:10.1007/3-540-44750-4_36}.'
```
Comment (valbert4) [inline codes/classical/q-ary_digits/frameproof.yml:107]: I think these can both be put in description.

File: codes/classical/q-ary_digits/traceability.yml [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+  # Not all fields are indexed by a dash
+  #transversal_gates: 'Transversal ... gates \cite{doi:ok-paper}. Comment out if doesn''t apply.'
+
+  #general_gates:
+    #- 'Universal gate set achieved by either additional ... gate.'
+    #- 'Magic-state distillation protocols'
+    #- 'kth Clifford hierarchy gates obtained by ... circuits'
+
+  #decoders:
```
Comment (valbert4) [inline codes/classical/q-ary_digits/traceability.yml:69]: Probably good to add a "decoder" field for the pirate decoder?

File: codes/classical/q-ary_digits/traceability.yml [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    #- 'Other fault-tolerant gadgets (measurements, encoders, error correcting steps)'
+    #- 'Noise-model-preserving gadgets, noise-biased gates, fault-tolerant flag error correction'
+    #- 'Pieceable fault tolerance.'
+
+  #code_capacity_threshold:
+    #- '\(1.5%\) error-correction threshold against some noise with *noiseless* decoder of some complexity \cite{arxiv:paper}.'
+    #- '\(5.2\%\) asymptotic upper bound from some estimation algortihm (Monte Carlo, e.g.) that may or may not have a feasible decoder.'
+
+  threshold:
+    - 'Suppose n is the number of users, k is the number of users known by the pirates, and p is the probability that the pirates cannot be traced. An open (public) resilient scheme using a hash function has the personal keys of the users consisting of O(k^{2}log n) descryption keys which is the amount of decryptions needed to reveal the information. The amount of data redundancy overhead is about O(k^{4}log n) \cite{doi:10.1109/18.841169}.'
```
Comment (valbert4) [inline codes/classical/q-ary_digits/traceability.yml:85]: This should all be in rate field.

File: codes/classical/q-ary_digits/traceability.yml [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+realizations:
+  # List and explain the different "domains" of realizations in list items.
+  # Group closely related realizations into a single item. It's OK to have a
+  # single list item. Each item should more or less correspond to what would be
+  # said in a paragraph.
+  - 'Code used in broadcast messages, pay-per-view movies, protecting copyrighted online material \cite{doi:10.1109/18.915661}.'
+
+notes:
+  - 'Fully resilient schemes guarantee to trace the source of any piracy \cite{doi:10.1109/18.841169}.'
+  - 'Threshold schemes have a parameter of the scheme, say q, for which the scheme only traces sources of pirate decoders that decrypt with greater probability than q \cite{doi:10.1109/18.841169}.'
```
Comment (valbert4) [inline codes/classical/q-ary_digits/traceability.yml:104]: @AlexanderBarg This "threshold" is different than that from quantum. Is this widespread notation? We can either put this in "rate" field, or make a new threshold-type field for this...

File: codes/classical/q-ary_digits/traceability.yml [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+  - 'Secret schemes are those that have their decryption scheme and keys as secret. The open schemes are those that the decryption schemes and decryption keys are public information \cite{doi:10.1109/18.841169}.'
+  - 'Note that in the feature section above, the hash function maps the users into a set of 2k^{2} decryption keys \cite{doi:10.1109/18.841169}.'
+  - 'Links to code tables \cite{doi:10.1109/18.841169} and \cite{doi:10.1109/18.915661}.'
+
+#relations: will put these in when they have been created
+  #parents:
+    #- code_id: ipp
```
Comment (valbert4) [inline codes/classical/q-ary_digits/traceability.yml:111]: Uncomment ipp

File: codes/classical/q-ary_digits/frameproof.yml [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+  - 'A length l q-ary c-frameproof code has at most \(tq^{\lceil l/c \rceil} + O(q^{\lceil l/c \rceil - 1})\) codewords where t is an integer between 1 and c and t is 1 mod c \cite{doi:10.1137/S0895480101384633}.'
+
+#relations:
+  #parents: #will add relations when IPP and traceability codes have been created and accepted into the zoo
```
Comment (AlexanderBarg) [inline codes/classical/q-ary_digits/frameproof.yml:111]: Cousins: Codes with the identifiable parent property, secure frameproof codes, separating codes, traceability codes.

File: codes/classical/q-ary_digits/traceability.yml [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  - 'Links to code tables \cite{doi:10.1109/18.841169} and \cite{doi:10.1109/18.915661}.'
+
+relations: 
+  #parents:
+    #- code_id: 
+      #detail: 
+  cousins:
+    - code_id: frameproof
+      detail: 'Frameproof codes fingerprint digital data and also they help prevent copyrighted information from unauthorized use \cite{arXiv:1411.5782}.'
+    - code_id: ipp
```
Comment (valbert4) [inline codes/classical/q-ary_digits/traceability.yml:51]: make ipp parent

File: codes/classical/q-ary_digits/frameproof.yml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+#notes:
+  
+relations:
+  #parents:
+    #- code_id: 
+     # detail: 'The smallest code family that includes this code that is defined over the same physical space structure or alphabet \cite{doi:10.1109/18.915661}.'
+  cousins:
+    - code_id: traceability
+      detail: 'Codes with traceability also deter users from unauthorized copying or sharing of digital materials by being able to trace source of leaks which also helps to ensure that an innocent user is not framed for the unauthorized activity \cite{doi:10.1109/18.841169}.'
+    - code_id: ipp
```
Comment (valbert4) [inline codes/classical/q-ary_digits/frameproof.yml:55]: make ipp parent

File: codes/classical/q-ary_digits/traceability.yml (hunk 10 above)
Comment (raleyr) [reply codes/classical/q-ary_digits/traceability.yml:51]: I thought I should but in one of the previous comments by Dr. Barg said "Cousins: Codes with the identifiable parent property, secure frameproof codes, separating codes, traceability codes." I want to confirm that I need to move ipp to parent

File: codes/classical/q-ary_digits/frameproof.yml (hunk 11 above)
Comment (raleyr) [reply codes/classical/q-ary_digits/frameproof.yml:55]: Same as above comment

File: codes/classical/q-ary_digits/traceability.yml (hunk 7 above)
Comment (AlexanderBarg) [reply codes/classical/q-ary_digits/traceability.yml:39]: Not so much widespread; I suggest to ignore.

File: codes/classical/q-ary_digits/traceability.yml (hunk 10 above)
Comment (AlexanderBarg) [reply codes/classical/q-ary_digits/traceability.yml:51]: confirmed

Comment (valbert4) [reply codes/classical/q-ary_digits/traceability.yml:51]: Thanks, I forgot. We need at least one parent though. Let's put ecc for now, for both of these.

**Pull request closed without merging**